Unit test TestTraceflow is flaky

**Describe the bug**
<!--
A clear and concise description of what the bug is.
If you believe this bug is a security issue, please don't use this template and follow our [security guidelines](/SECURITY.md)
-->
I met the following error twice:
```
I0121 03:52:57.587773   16303 controller.go:138] Starting TraceflowController
I0121 03:52:57.587997   16303 shared_informer.go:240] Waiting for caches to sync for TraceflowController
I0121 03:52:57.588053   16303 shared_informer.go:247] Caches are synced for TraceflowController 
I0121 03:52:58.592122   16303 controller.go:170] Processing Traceflow tf1 ADD event
E0121 03:52:58.592675   16303 controller.go:273] "Invalid Traceflow request" err="requested source Pod ns1/pod1 not found" request=&{TypeMeta:{Kind: APIVersion:} ObjectMeta:{Name:tf1 GenerateName: Namespace: SelfLink: UID:uid1 ResourceVersion: Generation:0 CreationTimestamp:2022-01-21 03:52:58.591825132 +0000 UTC m=+1.071151381 DeletionTimestamp:<nil> DeletionGracePeriodSeconds:<nil> Labels:map[] Annotations:map[] OwnerReferences:[] Finalizers:[] ClusterName: ManagedFields:[]} Spec:{Source:{Namespace:ns1 Pod:pod1 IP:} Destination:{Namespace:ns2 Pod:pod2 Service: IP:} Packet:{SrcIP: DstIP: Length:0 IPHeader:{SrcIP: Protocol:0 TTL:0 Flags:0} IPv6Header:<nil> TransportHeader:{ICMP:<nil> UDP:<nil> TCP:<nil>}} LiveTraffic:false DroppedOnly:false Timeout:2} Status:{Phase: Reason: StartTime:<nil> DataplaneTag:0 Results:[] CapturedPacket:<nil>}}
I0121 03:52:58.592847   16303 controller.go:176] Processing Traceflow tf1 UPDATE event
I0121 03:52:59.795128   16303 controller.go:170] Processing Traceflow tf2 ADD event
E0121 03:52:59.795713   16303 controller.go:273] "Invalid Traceflow request" err="using hostNetwork Pod as source in non-live-traffic Traceflow is not supported" request=&{TypeMeta:{Kind: APIVersion:} ObjectMeta:{Name:tf2 GenerateName: Namespace: SelfLink: UID:uid2 ResourceVersion: Generation:0 CreationTimestamp:2022-01-21 03:52:59.794937634 +0000 UTC m=+2.274263883 DeletionTimestamp:<nil> DeletionGracePeriodSeconds:<nil> Labels:map[] Annotations:map[] OwnerReferences:[] Finalizers:[] ClusterName: ManagedFields:[]} Spec:{Source:{Namespace:ns1 Pod:pod2 IP:} Destination:{Namespace:ns2 Pod:pod2 Service: IP:} Packet:{SrcIP: DstIP: Length:0 IPHeader:{SrcIP: Protocol:0 TTL:0 Flags:0} IPv6Header:<nil> TransportHeader:{ICMP:<nil> UDP:<nil> TCP:<nil>}} LiveTraffic:false DroppedOnly:false Timeout:2} Status:{Phase: Reason: StartTime:<nil> DataplaneTag:0 Results:[] CapturedPacket:<nil>}}
I0121 03:52:59.795907   16303 controller.go:176] Processing Traceflow tf2 UPDATE event
--- FAIL: TestTraceflow (2.31s)
    --- FAIL: TestTraceflow/normalTraceflow (1.00s)
        controller_test.go:105: 
            	Error Trace:	controller_test.go:105
            	Error:      	Expected value not to be nil.
            	Test:       	TestTraceflow/normalTraceflow
    --- FAIL: TestTraceflow/timeoutTraceflow (1.10s)
        controller_test.go:137: 
            	Error Trace:	controller_test.go:137
            	Error:      	Expected value not to be nil.
            	Test:       	TestTraceflow/timeoutTraceflow
        controller_test.go:140: 
            	Error Trace:	controller_test.go:140
            	Error:      	Should be true
            	Test:       	TestTraceflow/timeoutTraceflow
        controller_test.go:141: 
            	Error Trace:	controller_test.go:141
            	Error:      	Not equal: 
            	            	expected: "Invalid Traceflow request, err: requested source Pod ns1/pod1 not found"
            	            	actual  : "Traceflow timeout"
            	            	
            	            	Diff:
            	            	--- Expected
            	            	+++ Actual
            	            	@@ -1 +1 @@
            	            	-Invalid Traceflow request, err: requested source Pod ns1/pod1 not found
            	            	+Traceflow timeout
            	Test:       	TestTraceflow/timeoutTraceflow
I0121 03:52:59.895777   16303 controller.go:166] Shutting down TraceflowController
FAIL
coverage: 44.5% of statements
FAIL	antrea.io/antrea/pkg/controller/traceflow	2.389s
```
It should be caused by the same issue in fake clientset as the one in https://github.com/antrea-io/antrea/pull/1723 that resource creation events were missing if the resources were created in-between list and watch call of an informer.

--- a/pkg/controller/traceflow/controller_test.go
+++ b/pkg/controller/traceflow/controller_test.go
@@ -75,6 +75,11 @@ func TestTraceflow(t *testing.T) {
 	stopCh := make(chan struct{})
 	tfc.informerFactory.Start(stopCh)
 	tfc.crdInformerFactory.Start(stopCh)
+	// Must wait for cache sync, otherwise resource creation events will be missing if the resources are created
+	// in-between list and watch call of an informer. This is because fake clientset doesn't support watching with
+	// resourceVersion. A watcher of fake clientset only gets events that happen after the watcher is created.
+	tfc.informerFactory.WaitForCacheSync(stopCh)
+	tfc.crdInformerFactory.WaitForCacheSync(stopCh)
 	go tfc.Run(stopCh)
 
 	numRunningTraceflows := func() int {

